src/utils/EventEmitter.ts

```diff
@@ -8,24 +8,30 @@ function uuidv4(): string {
 
 /** Generic Event Emitter class following a subscribe/publish paradigm. */
 export class EventEmitter<T extends any[]> {
-  subscribers: Record<string, (...args: [...T]) => void | undefined> = {};
+  private subscribers: Map<string, (...args: [...T]) => void | undefined> = new Map();
+
+  constructor() {}
 
   subscribe(s: (...args: [...T]) => void): () => void {
     let uuid = uuidv4();
-    while (this.subscribers[uuid] !== undefined) uuid = uuidv4();
-    this.subscribers[uuid] = s;
+    while (this.subscribers.get(uuid) !== undefined) uuid = uuidv4();
+    this.subscribers.set(uuid, s);
 
     return () => {
-      delete this.subscribers[uuid];
+      this.subscribers.delete(uuid);
     };
   }
 
   emit(...args: [...T]): void {
-    for (const s in this.subscribers) {
-      const sub = this.subscribers[s];
-      if (sub === undefined) continue;
-
+    this.subscribers.forEach((sub) => {
```

Comment:
This is a case where I'll insist on the imperative for-of style for efficiency, because it's in a base library.